source/agora/consensus/protocol/Nominator.d

```diff
@@ -539,7 +535,11 @@ extern(D):
         auto next_dup = duplicate_value(&next_value);
         auto nextval = this.wrapValue(next_dup);
 
-        if (this.scp.nominate(slot_idx, nextval, this.empty_value))
+
+        // Use hash of previous block as the previous slot value for randomizing
+        auto prev_value = this.ledger.lastBlock.hashFull().serializeFull().toVec();
```

Comment:
Since this allocates on the C++ side, it would be nice if we could find a way to do it once per new block, not every nomination. Otherwise LGTM.

Reply:
Good idea, will update.

Reply:
Updated